backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlTypeUtils.java

```diff
@@ -89,21 +89,25 @@ private static void applyAutoid(Column c, Row row) {
     }
   }
 
-  private static void checkRequired(Column c, Row row) {
-    if (!row.isDraft()
-        && c.getComputed() == null
-        && !AUTO_ID.equals(c.getColumnType())
-        && c.isRequired()) {
-
-      if (c.isReference()) {
-        for (Reference r : c.getReferences()) {
-          if (row.isNull(r.getName(), r.getPrimitiveType())) {
+  private static void checkRequired(Column c, Row row, Map<String, Object> values) {
+    if (!row.isDraft() && c.getComputed() == null && !AUTO_ID.equals(c.getColumnType())) {
+      if (c.isRequired()) {
+        if (c.isReference()) {
+          for (Reference r : c.getReferences()) {
+            if (row.isNull(r.getName(), r.getPrimitiveType())) {
+              throw new MolgenisException("column '" + c.getName() + "' is required in " + row);
+            }
+          }
+        } else {
+          if (row.isNull(c.getName(), c.getColumnType())) {
             throw new MolgenisException("column '" + c.getName() + "' is required in " + row);
           }
         }
-      } else {
-        if (row.isNull(c.getName(), c.getColumnType())) {
-          throw new MolgenisException("column '" + c.getName() + "' is required in " + row);
+      } else if (c.isConditionallyRequired()) {
```

Comment:
I found 'isRequired' vs 'isConditionallyRequired' a bit confusing initially but it is growing on me now :-)